internal/sql/pg/system.go

```diff
@@ -198,6 +202,452 @@ var settingValidations = map[string]settingValidFn{
 	"server_encoding": wantStringFn("UTF8"),
 }
 
+func queryRowFunc(ctx context.Context, conn *pgx.Conn, sql string,
+	scans []any, fn func() error, args ...any) error {
+	rows, _ := conn.Query(ctx, sql, args...)
+	_, err := pgx.ForEachRow(rows, scans, fn)
```

Comment:
We should add better tests for scanning to different types, because I am fairly confident they will not work. For example, I don't think `Uint256Array`'s `Scan()` implementation will work, because it only checks for a `[]string` type, and I would expected pgx would give us either `[]any` or `[]pgtype.UUID`.

Reply:
> We should add better tests for scanning to different types, because I am fairly confident they will not work. For example, I don't think `Uint256Array`'s `Scan()` implementation will work, because it only checks for a `[]string` type, and I would expected pgx would give us either `[]any` or `[]pgtype.UUID`.

Yeah, I will add some tests using the scanners.  But an `sql.Scanner` will always be given a primitive type. That's pretty much the purpose of such a type.  (Note the the types in `pgtypes` are `sql.Scanner` themselves).  pgx is just smart enough to recognize when it's scanning into a `sql.Scanner` and skip past it's own "smart" codecs, so afaik we are sidestepping all that complexity and dealing with lcd types.

Will test this a bit.

For reference, the std lib docs expand on this:

```go
// Scanner is an interface used by [Rows.Scan].
type Scanner interface {
	// Scan assigns a value from a database driver.
	//
	// The src value will be of one of the following types:
	//
	//    int64
	//    float64
	//    bool
	//    []byte
	//    string
	//    time.Time
	//    nil - for NULL values
	//
	// An error should be returned if the value cannot be stored
	// without loss of information.
	//
	// Reference types such as []byte are only valid until the next call to Scan
	// and should not be retained. Their underlying memory is owned by the driver.
	// If retention is necessary, copy their values before the next call to Scan.
	Scan(src any) error
}
```